sympy/functions/special/singularity_functions.py

```diff
@@ -52,7 +52,7 @@ class SingularityFunction(Function):
     >>> diff(SingularityFunction(x, 4, 0), x, 2)
     SingularityFunction(x, 4, -2)
     >>> SingularityFunction(x, 4, 5).rewrite(Piecewise)
-    Piecewise(((x - 4)**5, x > 4), (0, True))
+    Piecewise(((x - 4)**5, x >= 4), (0, True))
```

Comment:
There is overlap between the two clauses. We could keep the old form by testing to see if n is positive; if so the old form can be used, and if not, the new form could be used (with the `>=` comparison). With the old form, if n==0 then when x = 4, the value of 0 would be returned instead of 1.

Reply:
Changes thus would be blocked by #25580

Reply:
So, we didn't quite have it correct in the first place?